### DISCUSSION TITLE
Automatic #regions for F# modules

### DISCUSSION THREAD

**Discussion by pkese** [ionide/ionide-vscode-fsharp#2001] [Ideas]
Latest [vscode release (#Minimap section headers)](https://code.visualstudio.com/updates/v1_88#_minimap-section-headers) got an option to specify `//#region ...` which then get reflected into Minimap section headers.

I often organize my code into nested F# modules rather than regions.

It would be nice to have an option for Ionide to implicitly present nested modules as Minimap sections (even without explicit #region definitions).

**Comment by baronfel** [comment]
We'd probably need to look at the implementation (https://github.com/microsoft/vscode/pull/190759) and figure out what part of our markers aren't being found.

Reading the PR, can you try using /* */ style block comments and seeing if those are recognized?

**Comment by Krzysztof-Cieslak** [comment]
As far as I understand the [PR code](https://github.com/dgileadi/vscode/blob/f5f7032550961ba8f638e2db01f0083aa72387ba/src/vs/editor/common/services/findSectionHeaders.ts#L51) this feature is based on the [folding section of language configuration](https://code.visualstudio.com/api/language-extensions/language-configuration-guide#folding)... which we don't define in [our language configuration](https://github.com/ionide/ionide-vscode-fsharp/blob/main/release/language-configuration.json)

**Reply by baronfel** [reply]
Interesting - so the implementation cannot use LSP-defined folding ranges? Boo hiss.

**Reply by Krzysztof-Cieslak** [reply]
I guess the idea is that it is specifically for things like `//#region` in which case static definition is good enough.

**Reply by Krzysztof-Cieslak** [reply]
The good side of things is that getting it enabled for F# should be fairly straightforward/
